### ISSUE TITLE
Publish page 500 error.

### ISSUE THREAD

**Issue by JostCrow** [divio/django-cms#4575]
When I tried to publish a page I got a 500 error. This had to do with the fact that the target (in treebeard was set to None).

AttributeError at /admin/cms/page/222/nl/publish/
'NoneType' object has no attribute 'is_descendant_of'

I created a temp fix for this in a treebeard fork. But this is not the corerct way. Because I think this is not really a treebeard bug.

What I found that the problem was is that the parent folder is not a leaf folder and the get_last_child function retured None. 

This was when I was publishing the first child. You can see what will happen in the treebeard code.
```Python
def update_move_to_child_vars(self):
        """Update preliminar vars in :meth:`move` when moving to a child"""
        newdepth = self.target.depth
        newpos = None
        siblings = []
        if self.pos in ('first-child', 'last-child', 'sorted-child'):
            # moving to a child
            parent = self.target
            newdepth += 1
            if self.target.is_leaf():
                # moving as a target's first child
                newpos = 1
                self.pos = 'first-sibling'
                siblings = get_result_class(self.node_cls).objects.none()
            else:
                self.target = self.target.get_last_child()
                self.pos = {
                    'first-child': 'first-sibling',
                    'last-child': 'last-sibling',
                    'sorted-child': 'sorted-sibling'}[self.pos]

            # this is not for save(), since if needed, will be handled with a
            # custom UPDATE, this is only here to update django's object,
            # should be useful in loops
            parent.numchild += 1

        return newdepth, siblings, newpos
```

**Comment by yakky** [comment]
@JostCrow which CMS version are you using?

**Comment by JostCrow** [comment]
I started with 3.1.2 and upgraded to 3.1.3 after the bug appeared.

On Mon, 12 Oct 2015 19:29 Iacopo Spalletti <notifications@github.com> wrote:

> @JostCrow <https://github.com/JostCrow> which CMS version are you using?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/divio/django-cms/issues/4575#issuecomment-147468845>.
>
-- 

Jorik Kraaikamp

**Comment by yakky** [comment]
Same underlying issue as #4576

**Comment by yakky** [comment]
@JostCrow would you try https://github.com/yakky/django-cms/tree/feature/treebeard?

**Comment by yakky** [comment]
@JostCrow tentative fix at #4717

**Comment by yakky** [comment]
Closed by #4717
